### PR TITLE
Revise PHP code style following by PSR-2 CSG rules.

### PR DESCRIPTION
**Pull request by guzzilar**
**1. Objective reason**

This PR's focusing on revise PHP code style by following the PSR-2 Coding Style Guide's rules only.
ref: http://www.php-fig.org/psr/psr-2

The list of code-revised are as follows
- Update 2 spaces 👉 4 spaces,
- Revise line break, space, etc. without touch or change any current code much (to avoid have to fix unit test, etc).
- Correct `if-else` control structure style

For example
1.
```php
// Wrong
if(...){
}

// Change to
if (...) {
}
```

2.
```php
// Wrong
if (!$a) {
}

// Change to
if (! $a) {
}
```

3.
```php
// Wrong
public function some() {
    // ...
}

// Change to
public function some()
{
    // ...
}
```

4.
```php
// Wrong
class Some {
    // ...
}

// Change to
class Some
{
    // ...
}
```

**2. Description of change**

Because this PR is only about update code style and LOC & Code Diff would totally hard to follow/review.
So, I decided to separate 1 commit per 1 file to make it (a bit) easy to follow the lines & review.

> note: Total file changed: 18 files

- 💄 code styling, lib/omise/OmiseAccount          3566971
- 💄 code styling, lib/omise/OmiseBalance          2644834
- 💄 code styling, lib/omise/OmiseCard         8858cb5
- 💄 code styling, lib/omise/OmiseCardList         b6d822f
- 💄 code styling, lib/omise/OmiseCharge           ac21fa9
- 💄 code styling, lib/omise/OmiseCustomer         21e2256
- 💄 code styling, lib/omise/OmiseDispute          b778c30
- 💄 code styling, lib/omise/OmiseRecipient            9123b15
- 💄 code styling, lib/omise/OmiseRefund           67b0b4c
- 💄 code styling, lib/omise/OmiseRefundList           3a792d5
- 💄 code styling, lib/omise/OmiseTest         8850e28
- 💄 code styling, lib/omise/OmiseToken            54c33de
- 💄 code styling, lib/omise/OmiseTransaction          3ec668d
- 💄 code styling, lib/omise/OmiseTransfer         b469f68
- 💄 code styling, lib/omise/res/OmiseVaultResource            8036d58
- 💄 code styling, lib/omise/res/OmiseApiResource          6fbf802
- 💄 code styling, lib/omise/res/obj/OmiseObject           3747249
- 💄 code styling, lib/omise/exception/OmiseExceptions         d63d8d4

Edit 1, add more commit
- 💄 code styling, update code comment block  d2cda74

**3. Users affected by the change**

None

**4. Impact of the change**

None

**5. Priority of change**

Normal

**6. Alternate solution**

None

### CODE REVIEW

File: lib/omise/OmiseTest.php [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
+    /**
+     * (non-PHPdoc)
+     *
+     * @see OmiseApiResource::getInstance()
+     */
+    public static function resource()
+    {
+        return parent::getInstance(get_class());
+    }
 }
```
Comment (nimid) [inline lib/omise/OmiseTest.php]: @guzzilar Please add a single blank line at the end of file, lib/omise/OmiseTest.php.

File: lib/omise/res/obj/OmiseObject.php [hunk 2]
```diff
@@ -1,83 +1,104 @@
...
-      $this->_secretkey = OMISE_SECRET_KEY;
+
+    /**
+     * Reload the object.
+     *
+     * @param array $values
+     */
+    public function refresh($values, $clear = false)
+    {
+        if($clear) {
```
Comment (nimid) [inline lib/omise/res/obj/OmiseObject.php]: @guzzilar Please add a space after control structure keyword, if.

File: lib/omise/OmiseTest.php (hunk 1 above)
Comment (guzzilar) [inline lib/omise/OmiseTest.php]: 👍  done

File: lib/omise/res/obj/OmiseObject.php (hunk 2 above)
Comment (guzzilar) [inline lib/omise/res/obj/OmiseObject.php]: 👍 done

File: lib/omise/res/OmiseApiResource.php [hunk 3]
```diff
@@ -7,265 +7,307 @@
...
+     *
+     * @param  string $clazz
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @throws Exception|OmiseException
+     *
+     * @return OmiseAccount|OmiseBalance|OmiseCharge|OmiseCustomer|OmiseToken|OmiseTransaction|OmiseTransfer
+     */
+    protected static function g_retrieve($clazz, $url, $publickey = null, $secretkey = null)
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please change the method name, `g_retrieve`, to be camelCase.

File: lib/omise/res/OmiseApiResource.php [hunk 4]
```diff
@@ -7,265 +7,307 @@
...
+     * @param  string $url
+     * @param  array  $params
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @throws Exception|OmiseException
+     *
+     * @return OmiseAccount|OmiseBalance|OmiseCharge|OmiseCustomer|OmiseToken|OmiseTransaction|OmiseTransfer
+     */
+    protected static function g_create($clazz, $url, $params, $publickey = null, $secretkey = null)
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please change the method name, `g_create`, to be camelCase.

File: lib/omise/res/OmiseApiResource.php [hunk 5]
```diff
@@ -7,265 +7,307 @@
...
-    $request_url = dirname(__FILE__).'/../../../tests/fixtures/'.$request_url.'-'.strtolower($requestMethod).'.json';
+    /**
+     * Updates the resource with the given parameters in an associative array.
+     *
+     * @param  string $url
+     * @param  array  $params
+     *
+     * @throws Exception|OmiseException
+     */
+    protected function g_update($url, $params)
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please change the method name, `g_update`, to be camelCase.

File: lib/omise/res/OmiseApiResource.php [hunk 6]
```diff
@@ -7,265 +7,307 @@
...
+    /**
+     * Destroys the resource.
+     *
+     * @param  string $url
+     *
+     * @throws Exception|OmiseException
+     *
+     * @return OmiseApiResource
+     */
+    protected function g_destroy($url)
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please change the method name, `g_destroy`, to be camelCase.

File: lib/omise/res/OmiseApiResource.php [hunk 7]
```diff
@@ -7,265 +7,307 @@
...
-      unset($request_dir[count($request_dir) - 1]);
-      $request_dir = implode('/', $request_dir);
+    /**
+     * Reloads the resource with latest data.
+     *
+     * @param  string $url
+     *
+     * @throws Exception|OmiseException
+     */
+    protected function g_reload($url)
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please change the method name, `g_reload`, to be camelCase.

File: lib/omise/res/OmiseApiResource.php [hunk 8]
```diff
@@ -7,265 +7,307 @@
...
+            $result = $this->_executeTest($url, $requestMethod, $key, $params);
+        } else {
+            $result = $this->_executeCurl($url, $requestMethod, $key, $params);
+        }
+
+        // Decode the JSON response as an associative array.
+        $array = json_decode($result, true);
+
+        // If response is invalid or not a JSON.
+        if (count($array) === 0 || ! isset($array['object']))
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please correct this inline control structure.

**Comment by guzzilar** [comment]
@nimid for that part (change method name to `gRetrieve`, `gCreate`, `gUpdate`, `gDestroy`, `gReload`) I have planned to do in another pr.

This PR wanna focus only update 
- 2 spaces 👉 4 spaces,
- revise line break, space, etc. without touch or change any current code much.

sry, I didn't mention about 👆 in the PR description, will add.

File: lib/omise/res/OmiseApiResource.php [hunk 9]
```diff
@@ -7,265 +7,307 @@
...
+
+        // Remove slash if it had in last letter.
+        $request_url = rtrim($request_url, '/');
+
+        // Finally.
+        $request_url = dirname(__FILE__).'/../../../tests/fixtures/'.$request_url.'-'.strtolower($requestMethod).'.json';
+
+        // Make a request from Curl if json file was not exists.
+        if (! file_exists($request_url)) {
+
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please delete this blank line.

File: lib/omise/res/OmiseApiResource.php [hunk 10]
```diff
@@ -7,265 +7,307 @@
...
+        // Config UserAgent
+        if (defined('OMISE_USER_AGENT_SUFFIX')) {
+            $options += array(CURLOPT_USERAGENT => $user_agent." ".OMISE_USER_AGENT_SUFFIX);
+        } else {
+            $options += array(CURLOPT_USERAGENT => $user_agent);
+        }
+
+        // Also merge POST parameters with the option.
+        if (count($params) > 0) $options += array(CURLOPT_POSTFIELDS => http_build_query($params));
```
Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Please correct this inline control structure.

File: lib/omise/res/OmiseApiResource.php (hunk 8 above)
Comment (guzzilar) [inline lib/omise/res/OmiseApiResource.php]: @nimid you meant, change
```php
if (count($array) === 0 || ! isset($array['object']))
    throw new Exception('Unknown error. (Bad Response)');
```
to
```php
if (count($array) === 0 || ! isset($array['object'])) {
    throw new Exception('Unknown error. (Bad Response)');
}
```
?

Comment (nimid) [inline lib/omise/res/OmiseApiResource.php]: @guzzilar Yes.

But if you only focus on space format, other style can be ignored.

Comment (guzzilar) [inline lib/omise/res/OmiseApiResource.php]: @nimid fine, this one not change much and no need to test/fix unit test

done 👍

File: lib/omise/res/OmiseApiResource.php [hunk 11]
```diff
@@ -7,265 +7,307 @@
...
+
+        // Decode the JSON response as an associative array.
+        $array = json_decode($result, true);
+
+        // If response is invalid or not a JSON.
+        if (count($array) === 0 || ! isset($array['object']))
+            throw new Exception('Unknown error. (Bad Response)');
+
+        // If response is an error object.
+        if ($array['object'] === 'error')
```
Comment (guzzilar) [inline lib/omise/res/OmiseApiResource.php]: 👍  done

File: lib/omise/res/OmiseApiResource.php (hunk 9 above)
Comment (guzzilar) [inline lib/omise/res/OmiseApiResource.php]: 👍 done

File: lib/omise/res/OmiseApiResource.php (hunk 10 above)
Comment (guzzilar) [inline lib/omise/res/OmiseApiResource.php]: 👍 done

**Comment by nimid** [comment]
@guzzilar Please provide me more information about this pull request.

Are there any request or issue related with this pull request?
Is this pull request related with issue #36?

**Comment by guzzilar** [comment]
https://github.com/omise/omise-php/pull/37#issuecomment-244845385

@nimid No, no request from anyone and no related with any issue.
The PR purpose is just wanna clean up code before implement new api / fix any issue.

**Comment by nimid** [comment]
@guzzilar Thank you.

My review for only space format at this round is finished.

**Comment by guzzilar** [comment]
updated PR's description

**Comment by guzzilar** [comment]
https://github.com/omise/omise-php/pull/37#issuecomment-244846282

@nimid thanks, if you find anything more you can ping me, I'll fix it in another PR 👍

File: lib/omise/OmiseCardList.php [hunk 12]
```diff
@@ -3,47 +3,57 @@
...
-   * retrieve
-   * @param stroing $id
-   * @return OmiseCard
-   */
-  public function retrieve($id) {
-    $result = parent::execute($this->getUrl($id), parent::REQUEST_GET, self::getResourceKey());
-    return new OmiseCard($result, $this->_customerID, $this->_publickey, $this->_secretkey);
-  }
+    /**
+     * retrieve
```
Comment (bank32) [inline lib/omise/OmiseCardList.php]: @nam `retrieve a card` is better ?

File: lib/omise/OmiseCharge.php [hunk 13]
```diff
@@ -3,88 +3,107 @@
...
-  public function reverse() {
-    $result = parent::execute(self::getUrl($this['id']).'/reverse', parent::REQUEST_POST, parent::getResourceKey());
-    $this->refresh($result);
+        return $this;
+    }
 
-    return $this;
-  }
+    /**
+     * list refunds
```
Comment (bank32) [inline lib/omise/OmiseCharge.php]: @nam I realize it may ridiculous but sometimes you have `.` sometimes it's not 😂

Comment (guzzilar) [inline lib/omise/OmiseCharge.php]: @bank32 it's not ridiculous thou, it's good 👍 
what do you think, should or should not?

Comment (bank32) [inline lib/omise/OmiseCharge.php]: @guzzilar definitely should thou.

File: lib/omise/OmiseCustomer.php [hunk 14]
```diff
@@ -3,94 +3,115 @@
...
-    if($this['object'] === 'customer' &&  !empty($options)) {
-      return new OmiseCardList($this['cards'], $this['id'], $options, $this->_publickey, $this->_secretkey);
-    } else if ($this['object'] === 'customer') {
-      return new OmiseCardList($this['cards'], $this['id'], $this->_publickey, $this->_secretkey);
+    /**
+     * Gets a list of all cards belongs to this customer.
+     *
+     * @return OmiseCardList
+     */
+    public function cards($options = array())
```
Comment (bank32) [inline lib/omise/OmiseCustomer.php]: @guzzilar you forgot the comment like this `@param  array  $options`.

File: lib/omise/OmiseCustomer.php [hunk 15]
```diff
@@ -3,94 +3,115 @@
...
-  /**
-   * cards() alias
-   * @deprecated deprecated since version 2.0.0 use '$customer->cards()'
-   * @return OmiseCardList
-   */
-  public function getCards($options = array()) {
-    return $this->cards($options);
-  }
+    /**
+     * cards() alias
```
Comment (bank32) [inline lib/omise/OmiseCustomer.php]: @guzzilar same as https://github.com/omise/omise-php/pull/37/files?diff=split#r78140765

File: lib/omise/OmiseTransfer.php [hunk 16]
```diff
@@ -2,80 +2,97 @@
...
-  /**
-   * Updates the transfer amount.
-   */
-  public function save() {
-    $this->update(array('amount' => $this['amount']));
-  }
+    /**
+     * Updates the transfer amount.
+     */
+    public function save()
```
Comment (bank32) [inline lib/omise/OmiseTransfer.php]: @guzzilar add comment `@return void` is would be better.

File: lib/omise/exception/OmiseExceptions.php [hunk 17]
```diff
@@ -1,120 +1,93 @@
...
-    return $this->_omiseError;
-  }
-}
 
-class OmiseAuthenticationFailureException extends OmiseException {
+    /**
+     * Returns an instance of an exception class from the given error response.
+     *
+     * @param  array $array
+     * @return OmiseAuthenticationFailureException|OmiseNotFoundException|OmiseUsedTokenException|OmiseInvalidCardException|OmiseInvalidCardTokenException|OmiseMissingCardException|OmiseInvalidChargeException|OmiseFailedCaptureException|OmiseFailedFraudCheckException|OmiseUndefinedException
```
Comment (bank32) [inline lib/omise/exception/OmiseExceptions.php]: @guzzilar need one more space above this line to be same as usual.

Comment (bank32) [inline lib/omise/exception/OmiseExceptions.php]: @guzzilar one more space above this line.

File: lib/omise/res/obj/OmiseObject.php [hunk 18]
```diff
@@ -1,83 +1,104 @@
...
+
+    // Omise secret key.
+    protected $_secretkey;
+
+    // Omise public key.
+    protected $_publickey;
+
+    /**
+     * Setup the Omise object. If no secret and public are passed the one defined
+     * in config.php will be used.
```
Comment (bank32) [inline lib/omise/res/obj/OmiseObject.php]: @guzzilar no need to break the line here.

File: lib/omise/res/obj/OmiseObject.php [hunk 19]
```diff
@@ -1,83 +1,104 @@
...
-      $this->_secretkey = $secretkey;
-    } else {
-      $this->_secretkey = OMISE_SECRET_KEY;
+
+    /**
+     * Reload the object.
+     *
+     * @param array $values
+     */
+    public function refresh($values, $clear = false)
```
Comment (bank32) [inline lib/omise/res/obj/OmiseObject.php]: @guzzilar lack of one parameter (`$clear`)

File: lib/omise/res/obj/OmiseObject.php [hunk 20]
```diff
@@ -1,83 +1,104 @@
...
+    {
+        if ($clear) {
+            $this->_values = array();
+        }
+
+        $this->_values = array_merge($this->_values, $values);
+    }
+
+    // Override methods of ArrayAccess
+    public function offsetSet($key, $value)
```
Comment (bank32) [inline lib/omise/res/obj/OmiseObject.php]: @guzzilar none comment about parameter description in this file.

File: lib/omise/res/obj/OmiseObject.php (hunk 18 above)
Comment (guzzilar) [inline lib/omise/res/obj/OmiseObject.php]: @bank32 to follow code comment style, do you mind if I'm not gonna change it.
like, https://github.com/omise/omise-php/pull/37/files#diff-7a0eacc7ad276bf5e64dca38d6b295b6R10

File: lib/omise/OmiseCardList.php (hunk 12 above)
Comment (guzzilar) [inline lib/omise/OmiseCardList.php]: done

File: lib/omise/OmiseCustomer.php (hunk 14 above)
Comment (guzzilar) [inline lib/omise/OmiseCustomer.php]: done

File: lib/omise/OmiseCustomer.php (hunk 15 above)
Comment (guzzilar) [inline lib/omise/OmiseCustomer.php]: fyi, skip for now

File: lib/omise/res/obj/OmiseObject.php (hunk 19 above)
Comment (guzzilar) [inline lib/omise/res/obj/OmiseObject.php]: done

File: lib/omise/res/obj/OmiseObject.php (hunk 20 above)
Comment (guzzilar) [inline lib/omise/res/obj/OmiseObject.php]: this one I'll create another PR to correct the code comment. thanks

**Comment by guzzilar** [comment]
@bank32 please check, I committed as your comment here https://github.com/omise/omise-php/pull/37/commits/9f76f8af2a4c7835e020f797737608c6a85bcd77

**Comment by bank32** [comment]
@guzzilar it's clear now 👍

**Comment by guzzilar** [comment]
thanks 👍
